lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -22,6 +24,31 @@ module LogStash; module Outputs; class ElasticSearch;
   TARGET_BULK_BYTES = 20 * 1024 * 1024 # 20MiB
 
   class HttpClient
+    # Gzip library from rails ActiveSupport
+    module Gzip
```

Comment:
Hi suyograo. Thank you for the tips, I never worked with neither ruby or es before, so they are really appreciated. I had a look at the manticore compression feature, but that is client-side. So, when compression is true (which is the case by default), it will send a header with "Accept-Encoding: gzip, deflate" to inform the server that it can handle both algos. Then, if the server knows about http compression, it can look at the accept-encoding and decide which algo it wants to use, and sends the data back with the response header "Content-Encoding: algo_chosen". So there is no harm always having the flag on, as the server decides to encode the data based on what the client knows. 

In our case, ElasticSearch app also implements encoding on receiving data on the bulk api, meaning that we have to act like the server would and set the Content-Encoding header to the algorithm we used and compress the content using the same algorithm. This could had been done with encoding-negociation, but I haven't managed to make es sending it's Accept-Encoding header.